app/src/main/java/protect/card_locker/ScanActivity.java

```diff
@@ -221,4 +231,34 @@ public void addFromImage(View view) {
             Log.e(TAG, "No activity found to handle intent", e);
         }
     }
+
+    private void showCameraPermissionMissingText(boolean show) {
+        customBarcodeScannerBinding.cameraPermissionDeniedLayout.cameraPermissionDeniedClickableArea.setOnClickListener(show ? null : v -> {
+            navigateToSystemPermissionSetting();
+        });
+        customBarcodeScannerBinding.background.setBackgroundColor(show ? getResources().getColor(R.color.transparent) : obtainThemeAttribute(R.attr.colorSurface));
```

Comment:
I believe you can just use the built-in `Color.TRANSPARENT` here:

```suggestion
        customBarcodeScannerBinding.background.setBackgroundColor(show ? Color.TRANSPARENT : obtainThemeAttribute(R.attr.colorSurface));
```

Then the new color value can be removed too.